src/flamenco/runtime/fd_executor.c

```diff
@@ -355,12 +355,13 @@ fd_executor_load_transaction_accounts( fd_exec_txn_ctx_t * txn_ctx ) {
     int   err      = fd_txn_borrowed_account_view_idx( txn_ctx, (uchar)i, &acct );
     ulong acc_size = err==FD_ACC_MGR_SUCCESS ? acct->const_meta->dlen : 0UL;
 
-    /* Try to collect rent on all writable accounts. If rent is collected
+    /* Try to collect rent on all writable accounts, except the fee payer (who's rent was already
+       collected within fd_executor_validate_transaction_fee_payer()). If rent is collected
        successfully, update the starting lamports accordingly to avoid
        unbalanced lamports issues during instruction execution.
        TODO: the rent epoch check in the conditional should probably be moved
        to inside fd_runtime_collect_rent_from_account. */
-    if( fd_txn_account_is_writable_idx( txn_ctx, (int)i ) && acct->const_meta->info.rent_epoch<=epoch ) {
+    if( i>0 && fd_txn_account_is_writable_idx( txn_ctx, (int)i ) && acct->const_meta->info.rent_epoch<=epoch ) {
```

Comment:
why dont you just loop starting at i=1

Reply:
initially bc of the account data size calculations, but upon second thought we can start at 1 cuz fee payer dlen should always be 0